### PR TITLE
docs: add keyboard control switch theme

### PR DESCRIPTION
**Pull request by dongwei1125**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

Use the tab key to quickly select switch, then use the enter key to switch theme.

### Before

before

### After

after

### CODE REVIEW

File: docs/.vitepress/vitepress/components/common/vp-theme-toggler.vue [hunk 1]
```diff
@@ -61,7 +64,7 @@ const beforeChange = (): Promise<boolean> => {
 </script>
 
 <template>
-  <div @click.stop="switchTheme">
+  <div @click.stop="switchTheme" @keydown.enter.stop="handleKeyEnter">
```
Comment (btea) [inline docs/.vitepress/vitepress/components/common/vp-theme-toggler.vue:67]: It should be possible to reuse `switchTheme` directly here.

Comment (dongwei1125) [reply docs/.vitepress/vitepress/components/common/vp-theme-toggler.vue:67]: > It should be possible to reuse `switchTheme` directly here.

Considered reusing `switchTheme` directly, but `switchTheme` event parameter is `MouseEvent`. 
That include the coordinates of clicking mouse and generate transition based on position of the coordinates.
And `keydown.enter` is an inappropriate `KeyboardEvent`.

Comment (btea) [reply docs/.vitepress/vitepress/components/common/vp-theme-toggler.vue:67]: You can expand the `switchTheme` function. The keyboard switching theme you currently implement does not apply the transition effect.

File: docs/.vitepress/vitepress/components/common/vp-theme-toggler.vue [hunk 2]
```diff
@@ -14,12 +14,12 @@ watch(
...
 
 let resolveFn: (value: boolean | PromiseLike<boolean>) => void
-const switchTheme = (event: MouseEvent) => {
+const switchTheme = (event: MouseEvent | KeyboardEvent) => {
   const isAppearanceTransition =
     // @ts-expect-error
     document.startViewTransition &&
     !window.matchMedia('(prefers-reduced-motion: reduce)').matches
-  if (!isAppearanceTransition || !event) {
+  if (!isAppearanceTransition || !event || event instanceof KeyboardEvent) {
```
Comment (btea) [inline docs/.vitepress/vitepress/components/common/vp-theme-toggler.vue:22]: The judgment here is not necessary. We should ensure that the transition effects are effective when switching themes using the keyboard and mouse.

**Comment by ntnyq** [comment]
Actually you can use key `Space` to toggle color mode even without this PR.

Refs:

- [[W3C - SCR35: Making actions keyboard accessible by using the onclick event of anchors and buttons](http://www.w3.org/TR/WCAG20-TECHS/SCR35)](https://www.w3.org/TR/WCAG20-TECHS/SCR35)

**Review by btea — APPROVED** [review]

**Review by tolking — APPROVED** [review]

**Pull request merged**